sound/soc/sof/core.c

```diff
@@ -92,6 +92,46 @@ void snd_sof_get_status(struct snd_sof_dev *sdev, u32 panic_code,
 }
 EXPORT_SYMBOL(snd_sof_get_status);
 
+/*
```

Comment:
@ranj063 another thing that I like to see in the commit message is the motivation for which we are switching to this state machine. What is wrong with the old implementation? It is not flexible enough? etc..

Reply:
@dbaluta I've added this to the commit message.